cpp/include/cudf/utilities/traits.hpp

```diff
@@ -279,6 +279,30 @@ constexpr inline bool is_integral()
  */
 bool is_integral(data_type type);
 
+/**
+ * @brief Indicates whether the type `T` is an integral type but not bool type.
+ *
+ * @tparam T  The type to verify
+ * @return true `T` is integral but not bool
+ * @return false  `T` is not integral or is bool
+ */
+template <typename T>
+constexpr inline bool is_integral_not_bool()
```

Comment:
Wow, I could have sworn we had this already. (I think I'm confusing this with a type-list in the CUDF tests.)

Thanks for adding this.